docs/lib/introduction.md

```diff
@@ -0,0 +1,789 @@
+
+```
+type Repository interface {
+   GetContext() Context
+
+   GetSpecification() RepositorySpec
+   ComponentLister() ComponentLister
+
+   ExistsComponentVersion(name string, version string) (bool, error)
+   LookupComponentVersion(name string, version string) (ComponentVersionAccess, error)
+   LookupComponent(name string) (ComponentAccess, error)
+
+   Close() error
+}
+```
+
+Again, within the ocm-lib, there are multiple implementations of this interface. But most importantly, the package
+[`genericocireg`](../../pkg/contexts/ocm/repositories/genericocireg) provides an implementation of the _OCM Repository
+interface_ that is based on the _OCI Repository interface_. In other words, this package provides a mapping of the
+_OCM Repository_ functionility to the _OCI Repository_ functionality. The
+[`ocireg`](../../pkg/contexts/ocm/repositories/ocireg) package and the [`ctf`](../../pkg/contexts/ocm/repositories/ctf)
+merely provide the functionality to convert, or rather wrap, their _OCI Repository Spec_ to an _OCM Repository Spec_
+(remember, spec types were pretty much serializable representations and factories for the actual type, as explained
+[here](introduction/context/registry/registry.go)).  
+For each _OCI Repository Type_, there also exists a corresponding _OCM Repository Type_.
+Thus, as there is an **oci** _repository type_ `OCIRegistry`, there is also an **ocm** _repository type_ `OCIRegistry`,
+as there is an **oci** _repository type_ `CommonTransportFormat`, there is also an **ocm** _repository type_
+`CommonTransportFormat`.
+Within the ocm-lib this is implemented through an **addition to the factory-based type registry** concept. In case the
+does not find an entry for a certain type (within the Decode method), this
+[implementation (defaultScheme)](../../pkg/runtime/scheme.go) of a factory-based type registry has a _fallback factory_
+(called `defaultdecoder`). For the OCM Repository Type Registry, this _fallback factory_ is usually
+[initialized](../../pkg/contexts/ocm/builder.go) with a Type Registry itself, the
+[`DefaultDelegationRegistry`](../../pkg/contexts/ocm/repositories/genericocireg/type.go) which is based on the
+[`delegationRegistry`](../../pkg/contexts/ocm/internal/delegation.go). This Type Registry's iterates through all its
+factories and checks whether they knew that type. Currently, the only factory known by this _fallback Type Registry_ is
+the one registered in the [`genericocireg`](../../pkg/contexts/ocm/repositories/genericocireg/type.go) package which
+uses the **oci** _repository types registry_ to perform the unmarshaling (thus, if I registered the type in the
+oci repository types registry, it is automatically available for the ocm repository registry).
+
+**Other Repository Types:**
+Besides _OCM_ and _OCI Repository Types_, there currently also exist a number of
+[_Credential Repository Types_](../../pkg/contexts/credentials/repositories) that have to implement the following
```

Comment:
I don't quite get this: If these Credential Repo Types already exist, what is then meant with "that have to implement the following interface" ? If they do exist, wouldn't that mean they have already implemented the interface ?